<launch>
  <arg name="mav_name" default="firefly"/>
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">   
    <arg name="world_name" value="$(find rotors_gazebo)/worlds/basic.world"/>
    <!--arg name="world_name" value="$(find rotors_gazebo)/worlds/powerplant.world"/-->

    <!-- <arg name="debug" value="true"/> -->
    <arg name="paused" value="true"/>
    <!-- <arg name="gui" value="false"/> -->
  </include>

  <group ns="$(arg mav_name)">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <arg name="model" value="$(find rotors_exercise)/urdf/firefly_fake_gps_sensor.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="$(arg log_file)"/>
    </include>

    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_exercise)/resource/lee_controller_$(arg mav_name)_soft.yaml" />
      <rosparam command="load" file="$(find rotors_exercise)/resource/$(arg mav_name).yaml" />

      <remap from="odometry" to="/firefly/ground_truth/odometry" />
    </node>

    <node name="hovering_example" pkg="rotors_gazebo" type="hovering_example" output="screen"/>

    <node name="waypoint_publisher_file" pkg="rotors_gazebo" type="waypoint_publisher_file" output="screen" args="$(find rotors_exercise)/resource/exercise1.txt"/>

    <node pkg="rotors_exercise" name="estimator_node" type="estimator_node" output="screen">
	 <param name="sigma_nx" value="1" />
	 <param name="sigma_nz" value="1" />
	 <param name="sigma_nu" value="0.004" />
    </node>
    <node pkg="rotors_exercise" name="path_node" type="path_node" output="screen" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rotors_exercise)/launch/display.rviz"/>

  </group>
</launch>
